dev/nfs/Makefile

```diff
@@ -0,0 +1,212 @@
+  $(WORKDIR)/config/dhcpd.conf \
+  $(WORKDIR)/config/tftpd-hpa \
+  $(WORKDIR)/provide/boot $(WORKDIR)/provide/root \
+  $(WORKDIR)/config/Dockerfile $(WORKDIR)/config/run.sh
+	docker build --tag "ot3-bootserver:latest" $(WORKDIR)/config
+
+.PHONY: apparmor-prep
+apparmor-prep: $(WORKDIR)/config/varfs-apparmor-profile
+	echo APPARMOR APPARMOR
+	sudo apparmor_parser -r -W $<
+
+.PHONY: check-route-exists
+check-route-exists:
+	$(if $(SPECROUTE),,$(warning $(noroutemsg)))
+
+.PHONY: check-route-not-default
+check-route-not-default:
+	$(if $(filter $(DEFROUTE),$(SPECROUTE)), $(error $(defaultroutemsg)))
+
+.PHONY: check-iface-route
+check-iface-route:
+	$(if $(filter $(DEFROUTE),$(DHCP_SUBNET_STUB).0),,$(warning $(badroutemsg)))
+
+.PHONY: do-setup-msg
+do-setup-msg:
+	$(info $(infomsg))
+
+.PHONY: check-image-exists
+check-image-exists:
+	$(if $(wildcard $(WORKDIR)/provide/root/*),,$(error $(noimgmsg)))
+
+.PHONY: check-no-hostservices
+check-no-hostservices:
+	$(if $(or $(shell ps -e | grep nfsd),$(shell ps -e | grep rpc[^i])),$(error $(hostservicesmsg)))
+
+.PHONY: check
+check: check-route-exists check-route-not-default check-iface-route check-no-hostservices check-image-exists do-setup-msg
+
+
+.PHONY: docker-run
+docker-run: stop docker-build $(if $(NOCHECK),,check) $(if $(filter yes,$(INSERT_APPARMOR)), apparmor-prep)
+	docker run                                                                     \
+	--name $(CONTAINER_NAME)                                                       \
+    --net host                                                                     \
+	-e NFS_LOG_LEVEL=DEBUG                                         \
+	-d                                                                             \
+	--cap-add SYS_ADMIN                                                            \
+    --cap-add SYS_MODULE                                                           \
+	--mount type=bind,src=$(WORKDIR)/provide,dst=/provide                          \
+    --mount type=bind,src=/lib/modules,dst=/lib/modules,readonly                   \
+	$(if $(filter yes,$(INSERT_APPARMOR)), --security-opt apparmor=erichough-nfs)  \
+	-p ${DHCP_SUBNET_STUB}.1::2049  -p ${DHCP_SUBNET_STUB}.1::2049/udp             \
+    -p ${DHCP_SUBNET_STUB}.1::111   -p ${DHCP_SUBNET_STUB}.1::111/udp              \
+    -p ${DHCP_SUBNET_STUB}.1::32765 -p ${DHCP_SUBNET_STUB}.1::32765/udp            \
+    -p ${DHCP_SUBNET_STUB}.1::32767 -p ${DHCP_SUBNET_STUB}.1::32767/udp            \
+    ot3-bootserver:latest $(IFACE)
+
+.PHONY: monitor
+monitor: docker-run
+	docker logs -f $(CONTAINER_NAME)
+
+# top level phony targets
+.PHONY: clean
+clean: stop
+	$(RM) -r $(WORKDIR)
+
+.PHONY: stop
+stop:
+	-docker stop $(CONTAINER_NAME)
+	-docker rm $(CONTAINER_NAME)
+
+
+# If an image was provided, copy it in; otherwise roll with what's present
+ifdef IMAGE
+.PHONY: set-image
+set-image: $(WORKDIR)
+	$(RM) -r $(WORKDIR)/unzip
+	mkdir -p $(WORKDIR)/unzip
```

Comment:
do we need `mkdir -p $(WORKDIR)/provide` under the recipe portion of this target?